### PR TITLE
Change ast assignment base

### PR DESCRIPTION
**Pull request by jlumpe**
Original PR to @bjodah's fork [here](https://github.com/bjodah/sympy/pull/15).

Removed `sympy.Relational` from class hierarchy as it doesn't really fit - they are structurally similar in that they have a left-hand side, a right-hand side, and a binary operation, but the semantics are very different. I doubt anyone has been directly making use of the fact that Assignment inherits from Relational, but this may break backward compatibility because the `rel_op` attribute has been changed to just `op`. Perhaps `rel_op` could be left in as an alias.

This removes the ~~fancy string printing~~ `repr()`, which makes it more consistent with other classes in the `ast` module. The pretty printing behavior has been retained. Doctests have been updated to match.

Also added common base class for both Assignment and AugmentedAssignment so it is easier to differentiate between them using isinstance().

NO ENTRY

### CODE REVIEW

File: sympy/printing/pretty/tests/test_pretty.py [hunk 1]
```diff
@@ -1087,79 +1084,6 @@ def test_pretty_relational():
     assert pretty(expr) in [ascii_str_1, ascii_str_2]
     assert upretty(expr) in [ucode_str_1, ucode_str_2]
 
-def test_Assignment():
```
Comment (asmeurer) [inline sympy/printing/pretty/tests/test_pretty.py]: Why were these tests removed? I think the pretty printing should be maintained, even if the class heirarchy changes.

**Comment by asmeurer** [comment]
@bjodah do you think we should get this in 1.2, or wait and put it in master? The changes here are less controversial than larger the class heirarchy rewrite that was proposed at 
https://github.com/bjodah/sympy/pull/15

File: sympy/codegen/tests/test_ast.py [hunk 2]
```diff
@@ -109,6 +97,29 @@ def test_AugAssign():
     raises(TypeError, lambda: aug_assign(A + A, '+', mat))
     raises(TypeError, lambda: aug_assign(B, '+', 0))
 
+
+def test_Assignment_printing():
```
Comment (jlumpe) [inline sympy/codegen/tests/test_ast.py]: @asmeurer Here are the new tests for `repr(Assignment)`, which follow the pattern of other AST classes. The old class inherited from `Relational` and so was printed like `x := y + 1`. The new class just inherits the string representation from the base AST class.

File: sympy/printing/pretty/tests/test_pretty.py (hunk 1 above)
Comment (jlumpe) [reply sympy/printing/pretty/tests/test_pretty.py]: The old class inherited from `Relational` and thus was printed like `x := y + 1`. The new class just inherits string printing behavior from the base AST class. I put the new tests for `repr(Assignment(...))` in `codegen/tests/test_ast.py`, which follows the pattern for the module.

Comment (asmeurer) [reply sympy/printing/pretty/tests/test_pretty.py]: I don't see why it shouldn't continue to pretty print with := with the new class structure.

**Comment by asmeurer** [comment]
The 1.2 release is out, so once https://github.com/sympy/sympy/pull/14838 is merged, change the base of this PR to `master`.

**Comment by bjodah** [comment]
I'm neutral to the pretty printing.
Why is this showing up as a big diff wrt. master? Looks like there should only be the last couple of commits in the PR showing up as new?

**Comment by asmeurer** [comment]
I think there is a bug in GitHub. As long as the hashes are the same as in master it is fine.

**Review by asmeurer — CHANGES_REQUESTED** [review]
The pretty printing needs to be fixed.

**Comment by jlumpe** [comment]
@asmeurer think this is finally ready to merge... I retained the original pretty-printing tests, which it now passes. However string printing and `repr()` is still more in line with the rest of the ast module.

**Review by asmeurer — APPROVED** [review]

**Pull request merged**